databricks_cli/dbfs/api.py

```diff
@@ -123,8 +134,48 @@ def get_file(self, dbfs_path, dst_path, overwrite, headers=None):
                 offset += bytes_read
                 local_file.write(b64decode(data))
 
+    def get_num_files_deleted(self, partial_delete_error):
+        message = partial_delete_error.response.json()["message"]
+        m = re.compile(r".*operation has deleted (\d+) files.*").match(message)
+        if not m:
+            raise ParseException(
+                "Unable to retrieve the number of deleted files from the error message: {}".format(
+                    message))
+        return int(m.group(1))
+
     def delete(self, dbfs_path, recursive, headers=None):
-        self.client.delete(dbfs_path.absolute_path, recursive=recursive, headers=headers)
+        num_consecutive_503_errors = 0
+        num_files_deleted = 0
+        while True:
+            try:
+                self.client.delete(dbfs_path.absolute_path, recursive=recursive, headers=headers)
+            except HTTPError as e:
+                if e.response.status_code == 503:
+                    try:
+                        error_code = e.response.json()['error_code']
+                    except (AttributeError, KeyError):
+                        error_code = None
+                    # Handle partial delete exceptions: retry until all the files have been deleted
+                    if error_code == DbfsErrorCodes.PARTIAL_DELETE:
+                        try:
+                            num_files_deleted += self.get_num_files_deleted(e)
+                            click.echo("\rDeleted {} files. Delete in progress...\033[K".format(
```

Comment:
hm, what are the \r and \033[K? Are they color codes? There are helpers to do this here: https://click.palletsprojects.com/en/7.x/api/#click.style

Reply:
`\r` moves the cursor at the beginning of the line and `\033[K` clears the remaining chars until the end of the line. Their purpose is to interactively update the number of delete files.